File: 2026-03-10/Logan L - Main/day10.py
input = open("input.txt", "r")
input = input.read()
pairs = { "(": ")", "[": "]", "{": "}", "<": ">" }
scores = { ")": 3, "]": 57, "}": 1197, ">": 25137 }
score = 0
for line in input.split("\n"):
    stack = []
    for char in line:
        if char in pairs:
            stack.append(char)
        else:
            if stack and pairs[stack[-1]] == char:
                stack.pop()
            else:
                score += scores[char]
                break
print(score)

File: 2026-03-10/Logan L - Main/day10.js
import fs from 'fs';
const input = fs.readFileSync('input.txt', 'utf8').trim().split(/\r?\n/);
const openToClose = { "(": ")", "[": "]", "{": "}", "<": ">" };
const errorScore = { ")": 3, "]": 57, "}": 1197, ">": 25137 };
let total = 0;
for (const line of input) {
  const stack = [];
  for (const ch of line) {
    if (openToClose[ch]) {
      stack.push(ch);
      continue;
    }
    const open = stack.pop();
    if (!open || openToClose[open] !== ch) {
      total += errorScore[ch] || 0;
      break;
    }
  }
}
console.log(`error score: ${total}`);
